Source/JavaScriptCore/runtime/ImportMap.cpp

```diff
@@ -34,9 +34,37 @@
 namespace JSC {
 namespace ImportMapInternal {
 static constexpr bool verbose = false;
+
+// https://html.spec.whatwg.org/C#merge-module-specifier-maps
+static void mergeModuleSpecifierMaps(ImportMap::SpecifierMap& oldMap, const ImportMap::SpecifierMap& newMap, const ImportMap::Reporter& reporter)
+{
+    // Instead of copying the maps and returning the copy, we're modifying the
+    // maps in place.
+    // 2. For each specifier → url of newMap:
+    for (auto& specifier : newMap.keys()) {
+        // 2.1. If specifier exists in oldMap, then:
+        if (oldMap.contains(specifier)) {
+            // 2.1.1. The user agent may report the removed rule as a warning to the
+            // developer console.
+            reporter.reportWarning(makeString("An import map rule for specifier '"_s, specifier, "' was removed, as it conflicted with an existing rule."_s));
+            // 2.1.2. Continue.
+            continue;
+        }
+        auto url = newMap.get(specifier);
+        // 2.2. Set mergedMap[specifier] to url.
+        oldMap.add(specifier, WTFMove(url));
+    }
+}
+
 }
 
-Expected<URL, String> ImportMap::resolveImportMatch(const String& normalizedSpecifier, const URL& asURL, const SpecifierMap& specifierMap)
+ImportMap::ImportMap(SpecifierMap&& imports, ScopesMap&& scopesMap, IntegrityMap&& integrity)
+    : m_imports(imports), m_scopesMap(scopesMap), m_integrity(integrity)
+{
+    initializeScopesVector();
```

Comment:
Since `initializeScopesVector` is not called from anywhere, we should have that code in this constructor instead of extracting it out as `initializeScopesVector` (since it may feel like it can be called from somewhere else).